File: backend/app/schemas/recent_view.py
# backend/app/schemas/recent_view.py
from datetime import datetime
from pydantic import BaseModel
from typing import Optional, List, Any, Dict


class RecentViewPerfumeItem(BaseModel):
    id: str
    name: str
    brand_name: Optional[str] = None
    image_url: Optional[str] = None
    gender: Optional[str] = None
    main_accords: Optional[List[str]] = None

    class Config:
        orm_mode = True


class RecentViewItem(BaseModel):
    viewed_at: Optional[datetime] = None
    perfume: RecentViewPerfumeItem

    class Config:
        orm_mode = True

File: lib/models/recent_view.dart
// lib/models/recent_view.dart
class RecentViewItem {
  final String perfumeId;
  final String name;
  final String brandName;
  final String? imageUrl;
  final DateTime viewedAt;

  RecentViewItem({
    required this.perfumeId,
    required this.name,
    required this.brandName,
    required this.imageUrl,
    required this.viewedAt,
  });

  factory RecentViewItem.fromJson(Map<String, dynamic> json) {
    return RecentViewItem(
      perfumeId: json["perfume_id"],
      name: json["name"],
      brandName: json["brand_name"],
      imageUrl: json["image_url"],
      viewedAt: DateTime.parse(json["viewed_at"]),
    );
  }
}
